### PR TITLE
[Policy] Update policyinsights to 0.4.0 package

### PR DESCRIPTION
**Pull request by pilor**
Update policyinsights python package to 0.4.0:

- Introduces a new "az policy metadata" resource type
- Adds a new --resource-discovery-mode parameter to "az policy remediation create"
- Increments the api-version used for "az policy event" and "az policy state"
- Updated and re-recorded tests

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by jsntcy** [comment]
@Juliehzl, @zhoxing-ms, please help review it, thanks.

**Comment by Juliehzl** [comment]
@pilor May I know what the difference between policy insight and policy? Typically we will use policy in package `azure-mgmt-resource` for `az policy` related commands.

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -166,8 +166,10 @@ Release History
 
 **Policy**
 
-* Support for Policy API version 2019-09-01.
+* Support for Policy API version 2019-09-01
```
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst]: Please rebase to dev and make your release note at the top of history.rst.

File: src/azure-cli/HISTORY.rst [hunk 2]
```diff
@@ -166,8 +166,10 @@ Release History
...
 * az policy set-definition: Support grouping within policy set definitions with `--definition-groups` parameter
+* Add new command `az policy metadata` to retrieve rich policy metadata resources
+* az policy remediation create: Specify whether compliance should be re-evaluated prior to remediation with the `--resource-discovery-mode` parameter
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst]: Could you add the ``  symbols on the command? like `az policy remediation create`

**Comment by pilor** [comment]
> @pilor May I know what the difference between policy insight and policy? Typically we will use policy in package `azure-mgmt-resource` for `az policy` related commands. If using azure-mgmt-resource package for policy, the version control should be https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L190

Policy entities (definition and assignments) are in azure-mgmt-resource.  Policy compliance querying is in policy insights.  They are separated into two different RPs mainly due to ownership and the fact that policy entities are in the somewhat special Microsoft.Authorization RP.  Policy insights has its own SDKs that are separate from azure-mgmt-resource

**Comment by pilor** [comment]
@juliehzl @zhoxing-ms thanks for the review.  Comments addressed

**Pull request merged**